libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -166,18 +170,25 @@ void CompilerUtils::storeInMemoryDynamic(Type const& _type, bool _padToWordBound
 		m_context << Instruction::DUP2 << Instruction::MSTORE;
 		m_context << u256(_padToWordBoundaries ? 32 : 24) << Instruction::ADD;
 	}
-	else
+	else // process all other types
 	{
 		unsigned numBytes = prepareMemoryStore(_type, _padToWordBoundaries);
-		if (numBytes > 0)
-		{
-			solUnimplementedAssert(
-				_type.sizeOnStack() == 1,
-				"Memory store of types with stack size != 1 not implemented."
-			);
-			m_context << Instruction::DUP2 << Instruction::MSTORE;
-			m_context << u256(numBytes) << Instruction::ADD;
-		}
+		solAssert(
+			numBytes > 0,
+			"Memory store of type " + _type.richIdentifier() + " not possible (stack size 0)."
+		);
+
+		solUnimplementedAssert(
```

Comment:
Please change this to a regular assert and move it prior to the call to `prepareMemoryStore`. You can also change the `else` into an `else if (_type.isValueType())` and then add an `else` case that only has `solAssert(false, ...)`.

Reply:
> You can also change [...]

Preferred, would be more consistent.